### DISCUSSION TITLE
How to serialize foreign keys in the returned content?

### DISCUSSION THREAD

**Discussion by wu-clan** [vitalik/django-ninja#453] [Q&A] [answered]
I checked [nested-objects](https://django-ninja.rest-framework.com/tutorial/response-schema/#nested-objects),
And it's easy to use when querying all the data,
But when I only query one piece of data, I use the same method, it can work.
But, My return format is fixed, so I can't specify the same format as the data response,
At this point, the content of the foreign key is only ID, not serialized.

Then I used models.Manager with natural_key, It seems that the serialization succeeded, but it's not what I want. He looks like this:
```python
{
    "pk": 1,
    "model": "store.book",
    "fields": {
        "name": "Mostly Harmless",
        "author": [
          "jiang",
          "liu"
    ],
    }
}
```

What can I do to get such data

```python
{
    "pk": 1,
    "model": "store.book",
    "fields": {
        "first_name": "jiang",
        "last_name": "liu",
        "birth_date": "1952-03-11",
    }
}
```

In fastapi, if I use a join query, it will be automatically serialized to the situation I want,
But for Django ninja, he can't

**Comment by SmileyChris** [comment]
Looks like you just need to read about [nested objects](https://django-ninja.rest-framework.com/tutorial/response-schema/#nested-objects) in your response schema.

**Comment by wu-clan** [comment, marked as answer]
As I described, it forced the use of `response=List[Schema]`, but he has only bare data

If I want to put the data in schema model, like:

```python
class Response(Schema)
    code: 200
    msg: str = 'Success'
    data: Any = None
```
like [demo](https://django-ninja.rest-framework.com/tutorial/response-schema/#nested-objects), i need to do this

```python
@api.get("/tasks")
def tasks(request):
    queryset = Task.objects.select_related("owner")
    return Response(data=serialize('json', queryset))
```
return will be like:

```python
{
  "code": 200,
  "msg": "Success"
  "data": [
      {
          "id": 1, 
          "title": "Task 1",
          "is_completed": false,
          "owner": 1  # < --- !!! not resolve
      },
      {
          "id": 2, 
          "title": "Task 2",
          "is_completed": false,
          "owner": null
      },
  ]
}
```

**Reply by SmileyChris** [reply]
That's not like the demo though. Don't try to do your own serialization.

What you'd want is:
```python
class OwnerSchema(Schema):
    first_name: str
    last_name: str
    birth_date: datetime.date

class TaskSchema(Schema):
    id: int
    title: str
    is_completed: bool
    owner: OwnerSchema

class TasksResponse(Schema):
    code = 200
    msg: str = "Success"
    data: list[TaskSchema]
```
```python
@api.get("/tasks")
def tasks(request):
    queryset = Task.objects.select_related("owner")
    return TasksResponse(data=list(queryset))
```

**Reply by wu-clan** [reply]
great
